src/snowflake/snowpark/_internal/packaging_utils.py

```diff
@@ -283,12 +286,13 @@ def identify_supported_packages(
 
             # Native packages should be anaconda dependencies, even if the requested version is not available.
             elif package_name in native_packages:
-                _logger.warning(
-                    f"Package {package_name}{version_text} contains native code, switching to latest available version "
-                    f"in Snowflake '{valid_packages[package_name][-1]}' instead."
-                )
+                if package_name not in package_dict:
+                    _logger.warning(
+                        f"Package {package_name}{version_text} contains native code, switching to latest available version "
+                        f"in Snowflake instead."
+                    )
+                    new_dependencies.append(Requirement.parse(package_name))
                 dropped_dependencies.append(package)
```

Comment:
Is it correct that we add an existing package to `dropped_dependencies`? I might miss some context so just want to do a sanity check.

Reply:
here the packages in the list `dropped_dependencies` are the packages that were downloaded by pip but need to be deleted from temp directory before the directory is zipped and uploaded to stage. `new_dependencies` is the list of packages available in snowflake that are dependencies of requested external packages.

Reply:
Makes sense. Thanks for the explanation. Maybe `dependencies_to_drop` is a better name :p

Reply:
I agree